### DISCUSSION TITLE
Issue in Two factor authentication and Forgot password Rest API in keycloak version 23.0.3

### DISCUSSION THREAD

**Discussion by Arunachalam6** [keycloak/keycloak#26594] [Q&A]
Hello there,
I’m currently facing an issue with 2FA and forgot password REST APIs that are not working properly.

https://{keycloak url}/auth/admin/realms/{realm name}/users/{userid}/execute-actions-email
For 2FA -
Headers => Authorization Token
Method => PUT
Request body => [“CONFIGURE_TOTP”]

Response: Status code= 403 (forbidden)
{
error: “unknown_error”
}

https://{keycloak url}/auth/admin/realms/{realm name}/users/{userid}/execute-actions-email
For Forgot password -
Headers => Authorization Token
Method => PUT
Request body => [“UPDATE_PASSWORD”]

Response: Status code= 403 (forbidden)
{
error: “unknown_error”
}

Kindly assist me in sorting this out.

**Comment by danielFesenmeyer** [comment]
For this endpoint the role realm-management role "MANAGE_USERS" is needed. Have you checked that the user triggering the email actually has this role?

**Reply by Arunachalam6** [reply]
Thanks for your response,

I have assigned the role -> ${role_manage-users} and checked, the email is not triggered.
req.body -> ["UPDATE_PASSWORD"]
response body ->
{
    "errorMessage": "Failed to send execute actions email"
}

**Reply by danielFesenmeyer** [reply]
Maybe your SMTP server is not correctly configured. Can you send a test mail from the corresponding realm settings page?
Maybe you can also find more details in the server logs.

**Reply by Arunachalam6** [reply]
The SMTP details also upto date configured, 
I have tested with the required actions like -> ["VERIFY_EMAIL"], ["UPDATE_PROFILE"]. The mail is sending when we use above required actions. 
I have also tested the test mail in authentication module in keycloak.

Once we give the [UPDATE_PASSWORD] the mail is not triggerring.
This is the issue, I have been facing.
